integr8ly/keycloak-operator
# INTLY-2987 - add users realmRoles reconciliation

## Motivation
JIRA: https://issues.jboss.org/browse/INTLY-2987
`realmRoles` of the users were not reconciled

## What
Renaming `KeycloakUserClientRole`type to -> `KeycloakUserRole` to re-use for realm roles as well.
Adding new method to keycloak realm phaseHandler - `reconcileUserRealmRoles`.
Other small fixes.

## Why
To enable us to set realmRoles via realm CR, e.g. `create-realm` or `admin`.

## How
Implementation is heavily inspired by `reconcileRolesForClient` and related methods in KC client.

## Verification Steps
fyi: I pushed current code to quay.io/omatskiv/keycloak-operator:v1.8.4


Create keycloak realm CR for `master` realm.
Make sure to set `spec.createOnly: true`
Add user with some `realmRoles` into `spec.users`, use non-default realmRoles, e.g. `create-realm`
Go to admin console and check that new user was created, and on "Role mappings" page realm role is in "Assigned" column.

## Checklist:
